File: #1232.check-if-it-is-a-straight-line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {

        int size = coordinates.size();
        if (size == 2) {
            return true;
        }

        vector<int> v1 = { coordinates[1][0] - coordinates[0][0],
                          coordinates[1][1] - coordinates[0][1] };
        vector<int> v2;
        for (int i = 2; i < size; i++) {
            v2 = {coordinates[i][0] - coordinates[0][0], coordinates[i][1] - coordinates[0][1]};
            if (v1[0]*v2[1] - v1[1]*v2[0] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: #1232.check-if-it-is-a-straight-line.js
/**
 * @param {number[][]} coordinates
 * @return {boolean}
 */
var checkStraightLine = function(coordinates) {
    if (coordinates.length === 2) {
        return true;
    }

    const [x1, y1] = coordinates[0];
    const [x2, y2] = coordinates[1];
    if (x2 - x1 === 0) {
        return coordinates.every(([x]) => x === x1);
    }

    const m = (y2 - y1) / (x2 - x1);
    const b = y1 - m * x1;
    return coordinates.every(([x, y]) => y === m * x + b);
};
